flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -104,22 +104,23 @@ public static StateBackend loadStateBackendFromConfig(
 
 		// by default the factory class is the backend name 
 		String factoryClassName = backendName;
+		final boolean logIsNotNull = logger != null;
 
 		switch (backendName.toLowerCase()) {
 			case MEMORY_STATE_BACKEND_NAME:
 				MemoryStateBackend memBackend = new MemoryStateBackendFactory().createFromConfig(config);
 
-				if (logger != null) {
+				if (logIsNotNull) {
```

Comment:
As you have changed this, should we add a test logic that is info log enabled?

Reply:
This is a minor change, i will add a test logic as you suggested, thanks

Reply:
I would suggest to revert the introduction of `logIsNotNull`. What is the benefit of the change?

Reply:
I just think `logIsNotNull` is more readable as there are many `logger != null` logic in the method. Ok, i will revert this point if it make no sense.